swap/src/api/request.rs

```diff
@@ -0,0 +1,679 @@
+                    .as_ref()
+                    .context("Could not get Bitcoin wallet")?;
+                let seller = self
+                    .params
+                    .seller
+                    .clone()
+                    .context("Parameter seller is missing")?;
+                let monero_receive_address = self
+                    .params
+                    .monero_receive_address
+                    .context("Parameter monero_receive_address is missing")?;
+                let bitcoin_change_address = self
+                    .params
+                    .bitcoin_change_address
+                    .clone()
+                    .context("Parameter bitcoin_change_address is missing")?;
+
+                let bitcoin_wallet = btc;
+                let seller_peer_id = seller
+                    .extract_peer_id()
+                    .context("Seller address must contain peer ID")?;
+                context
+                    .db
+                    .insert_address(seller_peer_id, seller.clone())
+                    .await?;
+
+                let behaviour = cli::Behaviour::new(
+                    seller_peer_id,
+                    env_config,
+                    bitcoin_wallet.clone(),
+                    (seed.derive_libp2p_identity(), context.config.namespace),
+                );
+                let mut swarm = swarm::cli(
+                    seed.derive_libp2p_identity(),
+                    context
+                        .config
+                        .tor_socks5_port
+                        .context("Could not get Tor SOCKS5 port")?,
+                    behaviour,
+                )
+                .await?;
+                swarm.behaviour_mut().add_address(seller_peer_id, seller);
+
+                tracing::debug!(peer_id = %swarm.local_peer_id(), "Network layer initialized");
+
+                let (event_loop, mut event_loop_handle) =
+                    EventLoop::new(swap_id, swarm, seller_peer_id)?;
+                let event_loop = tokio::spawn(event_loop.run());
+
+                let max_givable = || bitcoin_wallet.max_giveable(TxLock::script_size());
+                let estimate_fee = |amount| bitcoin_wallet.estimate_fee(TxLock::weight(), amount);
+
+                let (amount, fees) = match determine_btc_to_swap(
+                    context.config.json,
+                    event_loop_handle.request_quote(),
+                    bitcoin_wallet.new_address(),
+                    || bitcoin_wallet.balance(),
+                    max_givable,
+                    || bitcoin_wallet.sync(),
+                    estimate_fee,
+                )
+                .await
+                {
+                    Ok(val) => val,
+                    Err(error) => match error.downcast::<ZeroQuoteReceived>() {
+                        Ok(_) => {
+                            bail!("Seller's XMR balance is currently too low to initiate a swap, please try again later")
```

Comment:
Where are exceptions caught? Are they caught or will this cause the CLI to panic?

Reply:
If an error gets caught, the function will[ return early with the error](https://docs.rs/anyhow/latest/anyhow/macro.bail.html). So it's not a panic.

Reply:
Is the error returned to the API caller?

Reply:
Yes, the API methods returns a Result<value, error>